pkg/controller/endpointslicemirroring/reconciler.go

```diff
@@ -196,9 +197,15 @@ func (r *reconciler) reconcileByPortMapping(
 		// if >0 existing slices, mark all but 1 for deletion.
 		slices.toDelete = existingSlices[1:]
 
-		// Return early if first slice matches desired endpoints.
+		// generated slices must mirror all endpoints annotations but EndpointsLastChangeTriggerTime
+		compareAnnotations := cloneAndRemoveKey(endpoints.Annotations, corev1.EndpointsLastChangeTriggerTime)
+		compareLabelsTemp := cloneAndRemoveKey(existingSlices[0].Labels, discovery.LabelManagedBy)
+		compareLabels := cloneAndRemoveKey(compareLabelsTemp, discovery.LabelServiceName)
+		// Return early if first slice matches desired endpoints, labels and annotations
 		totals = totalChanges(existingSlices[0], desiredSet)
-		if totals.added == 0 && totals.updated == 0 && totals.removed == 0 {
+		if totals.added == 0 && totals.updated == 0 && totals.removed == 0 &&
+			apiequality.Semantic.DeepEqual(endpoints.Labels, compareLabels) &&
```

Comment:
we need apiequality, `reflect.Deepequal()` doesn't differentiate between `nil` and `map[string]string{}`